### PR TITLE
Add sample demonstrating startup script

### PR DESCRIPTION
**Pull request by fhinkel**
Add a sample that demonstrates how to create a VM
and install Apache and a custom homepage on startup.

Drive-by fix: Use lint config generated by repo-tools, do not use
Node 8 features like destructuring.

### CODE REVIEW

File: samples/startup-script/index.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    },
+  };
+
+  zone
+    .createVM(name, config)
+    .then(data => {
+      const vm = data[0];
+      const operation = data[1];
+
+      operation.on('complete', () => {
```
Comment (stephenplusplus) [inline samples/startup-script/index.js]: I believe you can return the operation and getMetadata call here, which should help keep the promise chain flat to catch errors:

```js
const vm = zone.vm(name);

vm
  .create(config)
  .then(data => {
    const operation = data[1];
    return operation.promise();
  })
  .then(() => {
    return vm.getMetadata();
  })
  .then(data => {
    const metadata = data[0];

    // ...
```

File: samples/startup-script/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "compute-sample-startup-script",
+  "version": "1.0.0",
+  "description": "Start a Google Compute Engine and run the startup script.",
+  "main": "index.js",
+  "dependencies": {
+    "@google-cloud/compute": "^0.10.0"
```
Comment (alexander-fenster) [inline samples/startup-script/package.json]: Oh. That way we need to change the samples-test task in CircleCI to run `npm link ../../' in this folder, to make sure it runs against the source code that is checked out and not the source code from NPM.

File: samples/startup-script/index.js (hunk 1 above)
Comment (fhinkel) [reply samples/startup-script/index.js]: Flattened. Thanks, looks much better this way.

File: samples/startup-script/package.json (hunk 2 above)
Comment (fhinkel) [reply samples/startup-script/package.json]: Thanks. Added an additional link to the `samples-test` script. Should I split it up in two scripts or is it OK for now?

**Comment by alexander-fenster** [comment]
I'll add a commit to your branch to check the samples and then revert it back. Please do not merge until I say 'OK' :)

**Comment by alexander-fenster** [comment]
Oh, it's not a branch, it's your repo. Sad :)  I wanted to temporary remove the branch filter 'master' from the samples_test task in `.circleci/config.yml` to verify it works in Circle, then put it back. Can you do it?

**Comment by alexander-fenster** [comment]
Alternatively, we can just merge and see if it fails! :)

File: package.json [hunk 3]
```diff
@@ -56,7 +56,7 @@
     "generate-scaffolding": "repo-tools generate all && repo-tools generate lib_samples_readme -l samples/ --config ../.cloud-repo-tools.json",
     "lint": "repo-tools lint --cmd eslint -- src/ samples/ system-test/ test/",
     "prettier": "repo-tools exec -- prettier --write src/*.js src/*/*.js samples/*.js samples/*/*.js test/*.js test/*/*.js system-test/*.js system-test/*/*.js",
-    "samples-test": "cd samples/ && npm link ../ && npm test && cd ../",
+    "samples-test": "cd samples/ && npm link ../ && cd startup-script && npm link ../../ && cd ../ && npm test && cd ../",
```
Comment (alexander-fenster) [inline package.json]: looks like here we won't run `npm test` in `samples/`, just in `samples/startup-script`. Maybe split this `samples-test` into two, as you did in `samples/package.json`?

Comment (alexander-fenster) [reply package.json]: nope, I was wrong.

**Comment by fhinkel** [comment]
Should I squash and merge and we see if it fails? It's a lot of new tools/configs for me. I probably missed one 😬 . 

Do you usually push to a branch in this repo instead of your own (so others can push changes and run the CI)?

**Comment by alexander-fenster** [comment]
I prefer developing in branch exactly for this reason: I can tweak the Circle configuration to run whatever I need in this current branch, and it will run system and samples tests for me. Then revert Circle config changes and merge to master.

But in this particular case I suggest that you just merge :)  (after you get a required approval on samples code - not sure if it's @jmdobry or anyone else who should give it, not me definitely!)  You merge, it either passes samples tests or fails, we fix then if needed. Not a huge problem.

**Comment by fhinkel** [comment]
I thought you're approval is enough, GitHub thinks it's enough 😄 Pinging @jmdobry 🙏

**Comment by alexander-fenster** [comment]
@fhinkel and you know what? All tests have passed! :)

**Comment by alexander-fenster** [comment]
https://circleci.com/workflow-run/34b268c2-caa4-4e44-a860-f1cf0ca6a687

**Pull request merged**